internal/epoch/epoch_manager.go

```diff
@@ -674,7 +674,7 @@ func (e *Manager) GetCompleteIndexSet(ctx context.Context, maxEpoch int) ([]blob
 }
 
 // WriteIndex writes new index blob by picking the appropriate prefix based on current epoch.
-func (e *Manager) WriteIndex(ctx context.Context, dataShards map[blob.ID]blob.Bytes) ([]blob.Metadata, error) {
+func (e *Manager) WriteIndex(ctx context.Context, dataShards map[blob.ID]blob.Bytes, retentionMode string, retentionPeriod time.Duration) ([]blob.Metadata, error) {
```

Comment:
Can the retention parameters be encapsulated and retained in the blob storage provider (`e.st`), instead of having to wire these parameters through for every call?

Reply:
I think we might be able to do that.

Reply:
I think there is some chicken-egg problem with the compactor method which still resides within ibm-v1.

However, my latest commit moves the retention params under the epoch manager with minimal changes, PTAL